mmdet/models/anchor_heads/yolact_head.py

```diff
@@ -0,0 +1,601 @@
+                        unmap)
+from mmdet.core.bbox import (PseudoSampler, assign_and_sample, bbox2delta,
+                             build_assigner)
+from ..losses import smooth_l1_loss
+from ..registry import HEADS
+from ..utils import ConvModule
+
+
+@HEADS.register_module
+class YolactHead(nn.Module):
+    """
+    An anchor-based head used in [1]_.
+
+    Very similar to RetinaHead but the has the concepts of extra_head_layers
+    and extra_layers. extra_head_layers defines the extra layers between the
+    backbone and the prediction head. And the extra_head_layers is shared among
+    all the prediction branches. extra_layers is as same as the stacked_convs
+    in RetinaHead.
+
+    References:
+        .. [1]  https://arxiv.org/pdf/1904.02689.pdf
+    """
+
+    def __init__(self,
+                 num_classes,
+                 in_channels,
+                 extra_layers=0,
+                 extra_head_layers=1,
+                 octave_base_scale=3,
+                 scales_per_octave=1,
+                 conv_cfg=None,
+                 norm_cfg=None,
+                 num_protos=32,
+                 loss_cls_weight=1.0,
+                 loss_bbox_weight=1.0,
+                 feat_channels=256,
+                 anchor_ratios=[0.5, 1.0, 2.0],
+                 anchor_base_sizes=[8, 16, 32, 64, 128],
+                 target_means=(.0, .0, .0, .0),
+                 target_stds=(1.0, 1.0, 1.0, 1.0),
+                 img_size=[550, 550]):
+        super(YolactHead, self).__init__()
+        self.extra_layers = extra_layers
+        self.extra_head_layers = extra_head_layers
+        self.octave_base_scale = octave_base_scale
+        self.scales_per_octave = scales_per_octave
+        self.conv_cfg = conv_cfg
+        self.norm_cfg = norm_cfg
+        self.num_protos = num_protos
+        self.loss_cls_weight = loss_cls_weight
+        self.loss_bbox_weight = loss_bbox_weight
+        octave_scales = np.array(
+            [2**(i / scales_per_octave) for i in range(scales_per_octave)])
+        anchor_scales = octave_scales * octave_base_scale
+
+        self.in_channels = in_channels
+        self.num_classes = num_classes
+        self.feat_channels = feat_channels
+        self.anchor_scales = anchor_scales
+        self.anchor_ratios = anchor_ratios
+        self.anchor_base_sizes = anchor_base_sizes
+        self.target_means = target_means
+        self.target_stds = target_stds
+        self.img_size = img_size
+
+        self.cls_out_channels = num_classes
+
+        self.fp16_enabled = False
+
+        self.anchor_generators = []
+        for anchor_base in self.anchor_base_sizes:
+            self.anchor_generators.append(
+                YolactAnchorGenerator(anchor_base, anchor_scales,
+                                      anchor_ratios))
+
+        self.num_anchors = len(self.anchor_ratios) * len(self.anchor_scales)
+        self._init_layers()
```

Comment:
Please rearrange above lines following the order of `__init__` function of [`AnchorHead`](https://github.com/open-mmlab/mmdetection/blob/master/mmdet/models/anchor_heads/anchor_head.py)